shaping/output.go

```diff
@@ -146,6 +146,11 @@ type Output struct {
 	// the output in order to render each run in a multi-font sequence in the
 	// correct font.
 	Face *font.Face
+
+	// VisualIndex is the visual position of this run within its containing line where
+	// 0 indicates the leftmost run and increasing values move to the right. This is
+	// useful for sorting the runs for drawing purposes.
+	VisualIndex int32
```

Comment:
Do you have a particular reason to choose `int32` over `int` ?

Reply:
I was trying to make the struct smaller. We should never need more than MAXINT32 runs in a single paragraph (2 billion runs is quite a few), so I don't think it makes sense to make the struct another 4 bytes larger on 64-bit systems by using `int`.

Reply:
Makes sense, thank you.